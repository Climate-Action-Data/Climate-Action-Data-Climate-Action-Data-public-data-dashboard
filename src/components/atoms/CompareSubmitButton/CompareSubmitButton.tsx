import { Button, Flex } from '@chakra-ui/react'
import { FC } from 'react'
import { CompareIcon } from '../CompareIcon/CompareIcon'
import { useTranslation } from 'react-i18next'

export interface CompareSubmitButtonProps {
  isDisabled?: boolean
  onClick: () => void
}

const CompareSubmitButton: FC<CompareSubmitButtonProps> = (props) => {
  const { isDisabled, onClick } = props
  const { t } = useTranslation(`search`)

  return (
    <Button
      variant="accentPrimary32"
      width="100%"
      minW={`187px`}
      height="32px"
      maxWidth="200px"
      onClick={onClick}
      isDisabled={isDisabled}
      fontSize="14px"
      data-testid="compare-submit-button"
    >
      <Flex alignItems={`center`} textColor={`black`}>
        {t(`projectCompare.compare`)}
      </Flex>
      <CompareIcon width="16px" height="16px" color={`black`} ml="4px" />
    </Button>
  )
}

export default CompareSubmitButton
