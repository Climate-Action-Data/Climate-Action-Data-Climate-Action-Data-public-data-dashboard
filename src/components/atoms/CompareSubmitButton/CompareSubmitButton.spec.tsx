import { render } from '@testing-library/react'
import CompareSubmitButton, { CompareSubmitButtonProps } from './CompareSubmitButton'

const DEFAULT_PROPS: CompareSubmitButtonProps = {
  isDisabled: true,
  onClick: () => jest.fn(),
}

it(`renders correctly`, () => {
  const { container } = render(<CompareSubmitButton onClick={DEFAULT_PROPS.onClick} isDisabled={DEFAULT_PROPS.isDisabled} />)
  expect(container).toMatchSnapshot()
})
