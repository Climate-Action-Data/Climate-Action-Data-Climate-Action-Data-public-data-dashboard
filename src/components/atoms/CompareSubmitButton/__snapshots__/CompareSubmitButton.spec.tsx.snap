// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    class="chakra-button css-1t7pqmu"
    data-testid="compare-submit-button"
    disabled=""
    type="button"
  >
    <div
      class="css-1o3wa6x"
    >
      Compare
    </div>
    <svg
      class="chakra-icon css-1g0edds"
      focusable="false"
      viewBox="0 0 32 32"
    >
      <path
        d="M14.9998 30.1541C14.7161 30.1541 14.4785 30.0585 14.287 29.8671C14.0956 29.6757 13.9998 29.4381 13.9998 29.1543V27.3338H7.07677C6.40326 27.3338 5.83317 27.1004 5.3665 26.6338C4.89984 26.1671 4.6665 25.597 4.6665 24.9235V7.07745C4.6665 6.40393 4.89984 5.83385 5.3665 5.36718C5.83317 4.90051 6.40326 4.66718 7.07677 4.66718H13.9998V2.84665C13.9998 2.56289 14.0956 2.32529 14.287 2.13385C14.4785 1.9424 14.7161 1.84668 14.9998 1.84668C15.2836 1.84668 15.5212 1.9424 15.7126 2.13385C15.9041 2.32529 15.9998 2.56289 15.9998 2.84665V29.1543C15.9998 29.4381 15.9041 29.6757 15.7126 29.8671C15.5212 30.0585 15.2836 30.1541 14.9998 30.1541ZM6.66647 24.0005H13.9998V15.2055L6.66647 24.0005ZM18.6665 27.3338V16.0005L25.3331 24.0005V7.07745C25.3331 6.97487 25.2904 6.88084 25.2049 6.79535C25.1194 6.70988 25.0254 6.66715 24.9228 6.66715H18.6665V4.66718H24.9228C25.5963 4.66718 26.1664 4.90051 26.6331 5.36718C27.0998 5.83385 27.3331 6.40393 27.3331 7.07745V24.9235C27.3331 25.597 27.0998 26.1671 26.6331 26.6338C26.1664 27.1004 25.5963 27.3338 24.9228 27.3338H18.6665Z"
        fill="black"
      />
    </svg>
  </button>
</div>
`;
